test-harness/tests/test_cache.py

```diff
@@ -668,3 +668,249 @@ def test_if_early_boot_stage_can_recover_from_a_bit_slow_backend(
             with GuardedSubprocess(v):
                 time.sleep(backend_request_timeout * 0.3 + 1)  # let it warm-up!
                 ping_mesos_agent(ar, superuser_user_header)
+
+
+class TestCacheMarathon:
+
+    def test_upstream_wrong_json(
+            self, nginx_class, mocker, superuser_user_header):
+        filter_regexp = {
+            "Cannot decode Marathon apps JSON: ": SearchCriteria(1, True),
+        }
+
+        ar = nginx_class()
+
+        # Send wrong non-json response content
+        mocker.send_command(endpoint_id='http://127.0.0.1:8080',
+                            func_name='apps_response_raw',
+                            aux_data=b"wrong response")
+
+        url = ar.make_url_from_path('/service/nginx-enabled/foo/bar/')
```

Comment:
Technically, this makes no difference here (we are setting borked response anyway), but still - can we please use `/nginx-alwaysthere`? The `/nginx-enabled` is reserved for task that is not enabled by default and some people reading the tests may find it confusing.